### PR TITLE
For #15931: Sort Downloads from newest to oldest

### PR DESCRIPTION
**Pull request by joc-a**
### Pull Request checklist

- [x] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [x] **Screenshots**: This PR includes screenshots or GIFs of the changes made or an explanation of why it does not
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features. In addition, it includes a screenshot of a successful [accessibility scan](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor&hl=en_US) to ensure no new defects are added to the product.

### To download an APK when reviewing a PR:
1. click on Show All Checks,
2. click Details next to "Taskcluster (pull_request)" after it appears and then finishes with a green checkmark,
3. click on the "Fenix - assemble" task, then click "Run Artifacts".
4. the APK links should be on the left side of the screen, named for each CPU architecture

Downloads are now sorted from newest to oldest based on **createdTime**.

This is how the downloads appear in the notifications on the phone (sorted from newest to oldest):
Screenshot_1603711537

This is how the downloads now appear on Fenix (the same order as above):
Screenshot_1603711558

### CODE REVIEW

**Comment by gabrielluong** [comment]
@joc-a Thank you for the contribution! Would you be able to post some screenshots for a ux-review?

File: app/src/main/java/org/mozilla/fenix/library/downloads/DownloadFragment.kt [hunk 1]
```diff
@@ -37,18 +37,20 @@ class DownloadFragment : LibraryPageFragment<DownloadItem>(), UserInteractionHan
...
-                it.value.filePath,
-                it.value.contentLength.toString(),
-                it.value.contentType,
-                it.value.status
-            )
-        }.filter {
-            it.status == DownloadState.Status.COMPLETED
-        }.filterNotExistsOnDisk()
+        val items = requireComponents.core.store.state.downloads.values
+            .sortedByDescending { it.createdTime } // sort from newest to oldest
```
Comment (Amejia481) [inline app/src/main/java/org/mozilla/fenix/library/downloads/DownloadFragment.kt:41]: Could we add a test to cover the new feature? :)

**Review by Amejia481 — CHANGES_REQUESTED** [review]
Thanks for the patch 👍🏽 
It looks great, a left one comment before approving ✅

**Comment by kglazko** [comment]
Can we wait to land this? This could potentially cause conflicts with the large Delete Downloads PR I am wrapping up.

File: app/src/main/java/org/mozilla/fenix/library/downloads/DownloadFragment.kt (hunk 1 above)
Comment (joc-a) [reply app/src/main/java/org/mozilla/fenix/library/downloads/DownloadFragment.kt:41]: Sure! I'll try my hand at writing a test.

Comment (joc-a) [reply app/src/main/java/org/mozilla/fenix/library/downloads/DownloadFragment.kt:41]: I don't have much experience writing tests, but I thought I'd give it a shot.

I added `createdTime` field to DownloadItem and tried with the following code I wrote in DownloadControllerTest:

`@Test
    fun `downloads are sorted from newest to oldest`() {
        val testList = state.items

        val expectedList = testList.sortedByDescending { it.createdTime }

        assertEquals(testList, expectedList)
    }`

But that didn't work. I'm very new to this so maybe I'm not approaching this the right way, but I'd be happy to get some guidance! 😊

Comment (Amejia481) [reply app/src/main/java/org/mozilla/fenix/library/downloads/DownloadFragment.kt:41]: @joc-a  [here](https://github.com/Amejia481/fenix/commit/4c2d5409e5b8433abbe20a50b8cb065876aeaad5) you can see an example. I would modify a bit `DownloadFragment` to make it testable.

**Comment by liuche** [comment]
If your changes addressed the comments, please tag the reviewer for another review!

In order for UX to sign off, please also add a screenshot.

Thanks for your help @joc-a !

File: app/src/main/java/org/mozilla/fenix/library/downloads/DownloadFragment.kt (hunk 1 above)
Comment (joc-a) [reply app/src/main/java/org/mozilla/fenix/library/downloads/DownloadFragment.kt]: @Amejia481 Done, thanks so much!

**Review by Amejia481 — APPROVED** [review]
This looks good to me, 
@kglazko do you still want to wait for your pr to land?

**Pull request merged**